fedimint-client/src/db.rs

```diff
@@ -441,6 +460,63 @@ pub fn get_core_client_database_migrations() -> BTreeMap<DatabaseVersion, CoreMi
         })
     });
 
+    // Migration to add outcome_time to OperationLogEntry
+    migrations.insert(DatabaseVersion(1), |mut ctx| {
+        Box::pin(async move {
+            let mut dbtx = ctx.dbtx();
+
+            // Read all OperationLogEntries using V0 format
+            let operation_logs = dbtx
+                .find_by_prefix(&OperationLogKeyPrefixV0)
+                .await
+                .collect::<Vec<_>>()
+                .await;
+
+            // Build a map from operation_id -> max_time of inactive state
+            let mut op_id_max_time = BTreeMap::new();
+
+            // Process inactive states
+            let inactive_states = dbtx
+                .find_by_prefix(&InactiveStateKeyPrefixBytes)
+                .await
+                .collect::<Vec<_>>()
+                .await;
```

Comment:
Best to not collect here, the way the code is written you can easily consume the stream with a `while let Some(_) = stream.next().await {` loop instead.

Reply:
done